### PR TITLE
[DO NOT MERGE] create a new 'rapids' conda environment instead of installing packages in the 'base' environment

### LINKED ISSUE #712: [BUG] Notebook tests failing on latest 24.10 nightlies
**Describe the bug**

Several notebook jobs are failing on 24.10 nightlies

```text
Error during notebook tests!
Errors during cugraph/algorithms/community/Community-Clustering.ipynb
Errors during cugraph/algorithms/community/Spectral-Clustering.ipynb
Errors during cuml/arima_demo.ipynb
Errors during cuml/forest_inference_demo.ipynb
Errors during cuml/kmeans_demo.ipynb
Errors during cuml/linear_regression_demo.ipynb
Errors during cuml/nearest_neighbors_demo.ipynb
Errors during cuml/random_forest_demo.ipynb
Errors during cuspatial/trajectory_clustering.ipynb
```

([build link](https://github.com/rapidsai/docker/actions/runs/11103412516/job/30866980386))

The logs don't contain much other detail.

**Steps/Code to reproduce bug**

Just run the `build` CI job against branch-24.10 at https://github.com/rapidsai/docker/actions/runs/11103412516.

**Expected behavior**

N/A

**Environment details (please complete the following information):**

N/A

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by jameslamb**
Fixes #712

Caused by https://github.com/rapidsai/build-planning/issues/56.

Nightly notebook runs have been failing here for the last couple days. It looks like the root cause is *"mismatched RAPIDS nightly versions, e.g. using a newer `cuml` with an older `libraft`, because `conda` refuses to install packages depending on `fmt>=11`"*.

This proposes fixing that by installing all packages into a new `rapids` environment in the images produced here, instead of into the `base` environment. See https://github.com/rapidsai/docker/issues/712#issuecomment-2384746639 for more details.

### Benefits of this change

Ensures that we can produce container images for 24.10, and all future releases as RAPIDS now pins to `fmt>=11.0.2,<12`.

Reduces the risk of similar conflicts the next time that `conda` / `mamba` (from `conda-forge`) and RAPIDS are using incompatible versions of `fmt`, `spdlog`, or any other dependencies they share.

### Costs of this change

This is a user-facing breaking change for anyone relying on these images having the RAPIDS packages in the `base` environment. Moving to the `base` environment was an explicit requirement of the `rapidsai/docker` overhaul done about 1.5 years ago in #539.

It'll require some changes to the RAPIDS deployment docs, for any places documenting the `base` environment, e.g.

https://github.com/rapidsai/deployment/blob/003524e074cf0df70730d63296618b12893b2e9c/source/examples/rapids-sagemaker-higgs/Dockerfile#L6-L10

### How I tested this

On an x86_64 machine with CUDA 12.2, built the `notebooks` image locally:

```shell
docker build \
    --build-arg CUDA_VER=11.8.0 \
    --build-arg PYTHON_VER=3.12 \
    --build-arg LINUX_DISTRO=ubuntu \
    --build-arg LINUX_DISTRO_VER=22.04 \
    --build-arg LINUX_VER=ubuntu22.04 \
    --build-arg RAPIDS_VER=24.10 \
    --target notebooks \
    -f ./Dockerfile \
    -t rapidsai/base:delete-me-local \
    context/
```

Ran it with not command / entrypoint specified. Saw Jupyter Lab come up successfully, and I was able to run the `cuml/arima_demo.ipynb` notebook end-to-end successfully 🎉 

```shell
docker run \
    --rm \
    --gpus "0,1" \
    -p 1234:8888 \
    -it rapidsai/base:delete-me-local
```

<details><summary>Confirmed that if you bypass the entrypoint, the libraries are still all found (click me)</summary>

```shell
docker run \
    --rm \
    --gpus "0,1" \
    --entrypoint="" \
    -it rapidsai/base:delete-me-local \
    python -c "import cudf; print(cudf.__version__)"

# 24.10.00a400

docker run \
    --rm \
    --gpus "0,1" \
    --entrypoint="" \
    -it rapidsai/base:delete-me-local \
    which jupyter
# /opt/conda/envs/rapids/bin/jupyter

docker run \
    --rm \
    --gpus "0,1" \
    --entrypoint="" \
    -it rapidsai/base:delete-me-local \
    jupyter labextension list

# /opt/conda/envs/rapids/share/jupyter/labextensions
#         jupyterlab_pygments v0.3.0 enabled OK (python, jupyterlab_pygments)
#         dask-labextension v7.0.0 enabled OK (python, dask_labextension)
#        ...
#
# Disabled extensions:
#     @jupyterlab/apputils-extension:announcements
```

</details>

### CODE REVIEW

**Comment by jameslamb** [comment]
The current state of this PR (https://github.com/rapidsai/docker/pull/713/commits/c836a9da3aedf72944922dca0fdbacbe662368e5) is at least sufficient to get all CI passing, which confirms that #712 was the root cause here (not issues with the notebooks or library code).

([build link](https://github.com/rapidsai/docker/actions/runs/11129096505/job/30925416302?pr=713))

But as we've been discussing on #712, it by itself would be an unacceptably large amount of breakage, and there might be ways to make this less disruptive: https://github.com/rapidsai/docker/issues/712#issuecomment-2386370049. I'll explore those next.

File: Dockerfile
Comment (jameslamb) [inline Dockerfile:104]: All of this is to make it *feel like* `conda activate rapids` was run even though it wasn't, in situations where you can't rely on the entrypoint script being run (described by @jacobtomlinson in https://github.com/rapidsai/docker/issues/712#issuecomment-2386370049).

It's a hack and could hopefully be reverted completely in RAPIDS 24.12 through some combination of the following:

* `conda` relaxing its `mamba` constraint and therefore its `fmt` constraint
  - *(e.g. `conda install -y -n base conda 'fmt>=11'` working)*
  - *(see https://github.com/rapidsai/docker/issues/712#issuecomment-2384746639)*
* RAPIDS removing its runtime dependency on `fmt` (https://github.com/rapidsai/build-planning/issues/104)

If reviewers agree with this approach, I'll write up a separate issue to track that work of reverting all of this.

---

NOTE: I'm intentionally not doing this in the `raft-ann-bench` images... those are expected to be used with explicit entrypoints, as far as I can tell, and I've modified all those entrypoints with the appropriate conda activation commands.

**Review by bdice — APPROVED** [review]
I'm fine with this if it works for the use cases that @jacobtomlinson defined in #712.

File: .github/workflows/test-notebooks.yml
Comment (jacobtomlinson) [inline .github/workflows/test-notebooks.yml:92]: With the environment hacking is this necessary?

**Review by jacobtomlinson — APPROVED** [review]
Thanks for digging into this @jameslamb. This seems like it should be a good workaround.

Once it has been merged and nightlies are pushed I'll check a few of the deployment places and verify things are working as expected.

File: .github/workflows/test-notebooks.yml
Comment (jameslamb) [reply .github/workflows/test-notebooks.yml:92]: Ah you're right, it shouldn't be! I'll try reverting it.

**Comment by jameslamb** [comment]
Put a `DO NOT MERGE` label on this... I'm hoping that https://github.com/conda-forge/mamba-feedstock/pull/253 will resolve the root cause and we won't need this PR at all.

Will come back and update in a bit, once this build hopefully publishes new `libmambapy` packages: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1044497&view=results

**Comment by jameslamb** [comment]
Very happy to say.... upstream changes made this unnecessary 😁 

details: https://github.com/rapidsai/docker/issues/712#issuecomment-2389484732

Thanks for the help everyone!

**Pull request closed without merging**
